src/prplmesh_getall.lua

```diff
@@ -68,4 +68,184 @@
 #
 ################################################################################
 --]]
-print("0; 1;")
+
+require("ubus")
+
+function path(str)
+    local res = str:gsub("(.*)%s.*$","%1")
+    print(res)
+    return res
+end
+
+
+function get_indexes(path)
+
+    local _ubus_connection
+    local devices
+    local instances
+    local result = {}
+
+    _ubus_connection = ubus.connect()
+    if not _ubus_connection then
+        print("Ubus sock conn fail")
+        return false
+    end
+
+    -- Retrieve output from list method via path from UBus
+    instance = _ubus_connection:call(path, "list", { })
+    if not instance then
+        print(path .. " not found in UBus")
+        return false
+    end
+
+    -- Fill up path instance indexes in table.
+    for k, v in pairs(instance["instances"]) do
+        result[k] = v["index"]
+        --print(" value: " .. tostring(result[k]))
+    end
+
+    _ubus_connection:close()
+
+    return result
+--get_indexes
+end
+
+
+function get_device_mmx_out(device_num)
+--TODO fix logic
+    local mmx_string = "0;"
+
+    if  device_num <= 0 then
+        print("Invalid devices amount")
+        return false
+    end
+
+    for i=1,device_num,1 do
+        mmx_string = mmx_string .. tostring(i) .. ";"
+    end
+
+    return mmx_string
+
+--get_device_mmx_out
+end
+
+
+function get_dm_table (path)
+
+    local tmp = {}
+    local parsed_args = {}
+
+    if not path then
+        print("Bad path.")
+        return false
+    end
+
+    for str in string.gmatch(path, "(.-){i}") do
+            table.insert(tmp, str)
+    end
+
+    for idx,val in pairs(tmp) do
+        if(idx == 1) then
```

Comment:
You may not use parentheses here

Reply:
It is not true, we can use parentheses. but they are useless, removed.